packages/modules/product/src/migrations/Migration20241125090957.ts

```diff
@@ -0,0 +1,173 @@
+import { Migration } from "@mikro-orm/migrations"
+
+export class Migration20241125090957 extends Migration {
+  async up(): Promise<void> {
+    this.addSql(
+      'alter table if exists "product" drop constraint if exists "product_status_check";'
+    )
+
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_category_parent_category_id" ON "product_category" (parent_category_id) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_category_path" ON "product_category" (mpath) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_category_handle_unique" ON "product_category" (handle) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_collection_handle_unique" ON "product_collection" (handle) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql('drop index if exists "IDX_product_image_deleted_at";')
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_image_deleted_at" ON "image" (deleted_at) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_image_url" ON "image" (url) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_tag_value_unique" ON "product_tag" (value) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_type_value_unique" ON "product_type" (value) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'alter table if exists "product" alter column "status" type text using ("status"::text);'
+    )
+    this.addSql(
+      "alter table if exists \"product\" add constraint \"product_status_check\" check (\"status\" in ('draft', 'proposed', 'published', 'rejected'));"
+    )
+    this.addSql(
+      'alter table if exists "product" alter column "status" set default \'draft\';'
```

Comment:
info: re apply check and add a default as draft, was not present before but make sense to have